File: V1P3/V1P3.go
/*
	El algoritmo recorre el arreglo, buscando X numero, cuando lo encuentra le aumenta uno a su frecuencia, una vez
	terminada ese frecuencia la compara con la anterior para ver si hay otra mayor y remplazarla.
*/

package main

func main() {

	a := [7]int{7, 8, 7, 9, 7, 8, 10} // Creamos el arreglo

	mejorValor := a[0]
	mejorFrecuencia := 0

	for i := 0; i < len(a); i++ { // Recorremos el arreglo
		frecuencia := 0

		for j := 0; j < len(a); j++ { // Buscamos cuantas veces aparece a[i]
			if a[i] == a[j] { // Si son iguales, contamos la aparición
				frecuencia++
			}
		}

		if frecuencia > mejorFrecuencia { // Si encontramos una frecuencia mayor
			mejorFrecuencia = frecuencia
			mejorValor = a[i]
		}
	}

	println("Mejor valor:", mejorValor)
	println("Frecuencia:", mejorFrecuencia)
}

File: V1P3/V1P3.rs
fn main() {

    let a = [7, 8, 7, 9, 7, 8, 10];

    let mut mejor_valor = a[0];
    let mut mejor_frecuencia = 0;

    for i in 0..a.len() { // Recorremos el arreglo
        let mut frecuencia = 0;

        for j in 0..a.len() { // Buscamos cuántas veces aparece a[i]
            if a[i] == a[j] { // Si son iguales, aumentamos la frecuencia
                frecuencia += 1;
            }
        }

        if frecuencia > mejor_frecuencia { // Si encontramos una frecuencia mayor
            mejor_frecuencia = frecuencia;
            mejor_valor = a[i];
        }
    }

    println!("Mejor valor: {}", mejor_valor);
    println!("Frecuencia: {}", mejor_frecuencia);
}